### PR TITLE
Added OAUTH for kafka

### PR DESCRIPTION
**Pull request by flyingcougar**
1. Changed config for oauth specific variables
2. Token is fetched using https://pkg.go.dev/golang.org/x/oauth2/clientcredentials
3. TokenCache is implemented

### CODE REVIEW

File: pkg/config/config.go [hunk 1]
```diff
@@ -10,9 +10,12 @@ import (
...
 
 type SASL struct {
-	Mechanism string
-	Username  string
-	Password  string
+	Mechanism string `yaml:"mechanism"`
+	Username  string `yaml:"username"`
+	Password  string `yaml:"password"`
+	ClientID  string `yaml:"clientID"`
+	Secret    string `yaml:"secret"`
```
Comment (birdayz) [inline pkg/config/config.go:17]: it's the client_secret right, so what about calling the field `ClientSecret` ?

Comment (flyingcougar) [reply pkg/config/config.go]: 👍

**Comment by flyingcougar** [comment]
@birdayz  need some help here.... tests are failing but i dont see any reason why this change could do that. I have run those on the main branch and are failing as well...

**Comment by birdayz** [comment]
i'll look into it probably this weekend :) should be fine when that fail is sorted out.

**Comment by flyingcougar** [comment]
look better now :)

**Pull request merged**